code/defines/procs/AStar.dm

```diff
@@ -237,9 +186,55 @@ proc/AStar(start,end,adjacent,dist,maxnodes,maxnodedepth = 30,mintargetdist,minn
 
 	return path
 
-//wrapper that returns an empty list if A* failed to find a path
-/proc/get_path_to(var/start,var/end,var/adjacent,var/dist,var/maxnodes,var/maxnodedepth = 30,var/mintargetdist,var/minnodedist,id=null, var/turf/exclude=null)
-	var/list/path = AStar(start, end, adjacent, dist, maxnodes, maxnodedepth, mintargetdist, minnodedist,id, exclude)
-	if(!path)
-		path = list()
-	return path
+/turf/proc/reachableAdjacentTurfs(atom, ID)
+	var/list/L = new()
+	var/turf/simulated/T
+	if(ID)
+		for(var/dir in cardinal)
+			T = get_step(src,dir)
+			if(!istype(T) || T.density)
+				continue
+			if(!LinkBlockedWithAccess(T, ID))
+				L.Add(T)
+	else
+		for(var/dir in cardinal)
+			if(dir & atmos_adjacent_turfs)
+				T = get_step(src,dir)
+				if(!istype(T))
+					continue
+				if(!LinkBlocked(atom, T))
+					L.Add(T)
+	return L
+
+/turf/proc/LinkBlocked(atom, turf/T)
+	if(istype(atom, /atom/movable))
+		for(var/obj/O in T)
+			if(!O.CanPass(atom, T, 1))
+				return 1
+		return 0
+	return 0
+
+/turf/proc/LinkBlockedWithAccess(turf/T, obj/item/weapon/card/id/ID)
+	var/adir = get_dir(src, T)
+	var/rdir = get_dir(T, src)
+	if(DirBlockedWithAccess(src, adir, ID))
+		return 1
+	if(DirBlockedWithAccess(T, rdir, ID))
+		return 1
+	for(var/obj/O in T)
+		if(O.density && !istype(O, /obj/machinery/door) && !(O.flags & ON_BORDER))
+			return 1
+	return 0
+
+/proc/DirBlockedWithAccess(turf/T, dir, ID)

```

Comment:
You're doubling the same proc that's already present in the unsorted.dm (l. 208) file in your codebase.
Same with ```LinkBlocked``` and ```LinkBlockedWithAccess```.